modernizer-maven-plugin/src/main/java/org/gaul/modernizer_maven_plugin/ModernizerMojo.java

```diff
@@ -431,6 +432,17 @@ private Outputer buildOutputer() throws MojoExecutionException {
         if (Objects.requireNonNull(outputFormat) == OutputFormat.CONSOLE) {
             return new LoggerOutputer(getLog(), violationLogLevel);
         } else if (outputFormat == OutputFormat.CODE_CLIMATE) {
+            // make sure the output directory exists
+            if (!Files.exists(Paths.get(baseDir))) {
```

Comment:
This has a lot of calls to `Paths.get`.  Perhaps it would be better to convert at the top:
```
Path baseDir = Paths.get(project.getBuild().getDirectory());
```

And change the constructor to `CodeClimateOutputer` to accept a `Path` as well?